src/Session.php

```diff
@@ -1591,6 +1591,17 @@ public static function validateCSRF($data)
     public static function checkCSRF($data)
     {
 
+        $message = __("The action you have requested is not allowed.");
+        $requestToken = $data['_glpi_csrf_token'];
+        if (
+            isset($_SESSION['glpicsrftokens'][$requestToken])
+            && ($_SESSION['glpicsrftokens'][$requestToken] < time())
+        ) {
+            $link = $_SERVER['HTTP_REFERER'] ?? '';
+            $message = __("Your session has expired.") . "<br>";
+            $message .= sprintf(__("<a href='%s'>Click here</a> to go back to the previous page."), $link);
```

Comment:
Translations must not contain HTML tags. You have to handle it in a different way.

Also, maybe this link should always be present, even if session has not expired.